File: octal_to_decimal/CPP/src/oct_to_dec.cpp
#include "../include/oct_to_dec.hpp"
#include <cmath>

int OctToDecimal::octToDecimal(std::string octal) {
    int decimalNumber = 0;
    int octalDigit, position = 0;
    
    // Iterate through the octal string in reverse order
    for (int i = octal.length() - 1; i >= 0; --i) {
        // Convert character to integer
        octalDigit = octal[i] - '0';
        
        // Validate octal digit
        if (octalDigit < 0 || octalDigit > 7) {
            // Handle invalid input
            // You can throw an exception, return an error code, or handle it as appropriate for your application
        }
        
        // Convert octal digit to decimal and add it to the result
        decimalNumber += octalDigit * static_cast<int>(pow(8, position));
        ++position;
    }
    
    return decimalNumber;
}

File: octal_to_decimal/JAVA/src/oct_to_dec.java
public class oct_to_dec {
    public int convertToDecimal(String octal) {
        // Check if octal string is null
        if (octal == null) {
            throw new IllegalArgumentException("Input octal string cannot be null");
        }

        // Convert octal to decimal
        int decimal = 0;
        int n = octal.length();
        for (int i = 0; i < n; i++) {
            char digit = octal.charAt(i);
            int digitValue;
            if (digit >= '0' && digit <= '7') {
                digitValue = digit - '0';
            } else {
                // Invalid octal digit
                throw new IllegalArgumentException("Invalid octal digit: " + digit);
            }
            decimal = decimal * 8 + digitValue;
        }
        return decimal;
    }
}
